cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -215,5 +231,80 @@ Status ParquetFileFormat::OpenReader(
   return Status::OK();
 }
 
+using parquet::arrow::FromParquetStatistics;
+
+static std::shared_ptr<Expression> ColumnChunkStatisticsAsExpression(
+    const Field& field, const parquet::ColumnChunkMetaData& metadata) {
+  // For the remaining of this function, failure to extract/parse statistics
+  // are ignored by returning the `true` scalar. The goal is two fold. First
+  // avoid that an optimization break the computation. Second, allow the
+  // following columns to maybe succeed in extracting column statistics.
+
+  auto field_expr = field_ref(field.name());
+
+  // In case of missing statistics, return nothing.
+  if (!metadata.is_stats_set()) {
+    return scalar(true);
+  }
+
+  auto statistics = metadata.statistics();
+  if (statistics == nullptr) {
+    return scalar(true);
+  }
+
+  // Optimize for corner case where all values are nulls
+  if (statistics->num_values() == statistics->null_count()) {
+    std::shared_ptr<Scalar> null_scalar;
+    if (!MakeNullScalar(field.type(), &null_scalar).ok()) {
+      // MakeNullScalar can fail for some nested/repeated types.
+      return scalar(true);
+    }
+
+    return equal(field_expr, scalar(null_scalar));
+  }
+
+  // Nothing to infer about this column
```

Comment:
I think this is redundant; doesn't FromParquetStatistics already check for statistics with no min/max?